tools/github-event-processor/YmlAndConfigFiles/event-processor.yml

```diff
@@ -0,0 +1,95 @@
+name: GitHub Event Processor
+
+on:
+  issues:
+    types: [edited, labeled, opened, reopened, unlabeled]
+  # issue_comment is used for both issues and pull_requests
+  # github.event.issue.pull_request will be non-null on pull request comments
+  issue_comment:
+    types: [created]
+  # synchronize is the pull_request_target event when changes are pushed
+  # pull request merged is the closed event with github.event.pull_request.merged = true
+  pull_request_target:
+    types: [closed, labeled, opened, reopened, review_requested, synchronize, unlabeled]
+  pull_request_review:
+    types: [submitted]
+
+# This removes all unnecessary permissions, the ones needed will be set below.
+# https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token
+permissions: {}
+
+jobs:
+  event-handler:
+    permissions:
+      issues: write
+      pull-requests: write
+      # For OIDC auth
+      id-token: write
+      contents: read
+    name: Handle ${{ github.event_name }} ${{ github.event.action }} event
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: 'Az CLI login'
+        if: ${{ github.event_name == 'issues' && github.event.action == 'opened' }}
+        uses: azure/login@v1
+        with:
+            client-id: ${{ secrets.AZURE_CLIENT_ID }}
+            tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+            subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: 'Run Azure CLI commands'
+        if: ${{ github.event_name == 'issues' && github.event.action == 'opened' }}
+        run: |
+          LABEL_SERVICE_API_KEY=$(az keyvault secret show \
+            --vault-name issue-labeler \
+            -n issue-labeler-func-key \
+            -o tsv \
+            --query value)
+
+          echo "::add-mask::$LABEL_SERVICE_API_KEY"
+          echo "LABEL_SERVICE_API_KEY=$LABEL_SERVICE_API_KEY" >> $GITHUB_ENV
+
+      # To run github-event-processor built from source, for testing purposes, uncomment everything
+      # in between the Start/End-Build From Source comments and comment everything in between the
+      # Start/End-Install comments
+      # Start-Install
+      - name: Install GitHub Event Processor
+        run: >
+          dotnet tool install
+          Azure.Sdk.Tools.GitHubEventProcessor
+          --version 1.0.0-dev.20230303.6
```

Comment:
We should think about some way to share this version number between the different yml files so they stay in-sync. Perhaps some sort of reusable github workflow.